2024/XenSummit/Challenges_and_Status_of_Enabling_TrenchBoot_in_Xen_Hypervisor.md

```diff
@@ -0,0 +1,234 @@
+- Twitter [@TrenchBoot](https://twitter.com/trenchboot)
+
+???
+
+TrenchBoot is a framework that allows individuals and projects to build
+security engines to perform launch integrity actions for their systems. The
+framework builds upon Boot Integrity Technologies (BITs) that establish one or
+more Roots of Trust (RoT) from which a degree of confidence that integrity
+actions were not subverted is derived.
+
+The first TrenchBoot D-RTM implementation has been made for GRUB+Linux
+
+---
+
+# TrenchBoot vs TrustedBoot
+
+.left-column50[
+
+### TrustedBoot
+
+- TrustedBoot (tboot) supports only Intel TXT
+- Is an exokernel and Xen (or any other kernel) has to be aware of its
+  presence
+]
+.right-column50[
+
+### TrenchBoot
+
+- Aims for unified approach supporting both AMD and Intel
+  processors
+- The goal is to implement a native support for D-RTM to let Xen have full
+  control without any exokernels
+]
+
+???
+
+Some may know about tboot - Intel TXT reference implementation
+
+With native support for DRTM, there are some additional steps to be performed
+before the CPUs may be initialized (wait for SIPI state). There is also a
+matter of handling power state transitions and event log.
+
+---
+
+# Usecase: Qubes OS Anti Evil Maid
+
+.center[.image-30[![](/img/qubes-logo-home.svg)]]
+
+- Qubes OS Anti Evil Maid (AEM) is a set of software packages and utilities to
+  aid against [Evil Maid attacks](https://theinvisiblethings.blogspot.com/2009/10/evil-maid-goes-after-truecrypt.html)
+- Leverages DRTM and TPM to seal secrets, which are used by the owner to
+  confirm whether the device has been tampered with or not
+- Initially only Intel TXT and TPM1.2 with tboot was supported using legacy
+  BIOS boot mode, now TrenchBoot replaces tboot, extending the support to
+  TPM2.0 and AMD SKINIT
+
+---
+
+# Demo: Qubes OS Anti Evil Maid
+
+.center[ <a href="https://asciinema.org/a/661348?cols=96&rows=24"
+target="_blank"> <img src="https://asciinema.org/a/661348.svg" width="400px"/>\
+</a> ]
+
+.footnote[
+Watch on [asciinema](https://asciinema.org/a/661348?cols=96&rows=24)
+]
+
+---
+
+# Current status of the work
+
+- Added TPM 2.0 support for dom0 kernel and initrd measurements
+- Added eventlog extraction and parsing
+- Rebased changes onto the most recent TrenchBoot support patches to Linux
+  kernel introducing a unified TrenchBoot boot protocol
+- Final shape of patches is not yet known, the work on finalizing the boot
+  protocol including UEFI is pending on the [Linux kernel mailing list,
+  currently patchset
+  v8](https://lore.kernel.org/lkml/8d543a15-af62-4403-b2e0-3b395edfe9e4@amd.com/T/)
+- Simplified installation of the required packages by providing a RPM
+  repository for QubesOS
+- Implemented support for booting with Intel TXT and TPM 2.0 in legacy BIOS
+  boot mode
+- Implemented support for booting with AMD SKINIT and TPM 1.2/2.0 in legacy
+  BIOS boot mode
```

Comment:
I would present it based on what @macpijan showed on Qubes OS Summit 2023; we need roadmaps, graphics, and updates, and how much was finished out of that thing we planned. If there were changed assumptions during security, we should also inform the community about that.